aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/internal/crypto/ContentCryptoScheme.java

```diff
@@ -198,6 +198,34 @@ CipherLite createCipherLite(SecretKey cek, byte[] iv, int cipherMode,
         }
     }
 
+    private Cipher createCipher(Provider provider, boolean alwaysUseProvider)
+            throws GeneralSecurityException {
+
+        String algorithm = getCipherAlgorithm();
+        String preferredProvider = getPreferredCipherProvider();
+
+        // If the user has specified that they always want to use the provider they
+        // specified, that wins (this is not the default for backwards compatibility
+        // reasons).
+        if (alwaysUseProvider) {
+            return Cipher.getInstance(algorithm, provider);
+        }
+
+        // Otherwise, if this crypto scheme prefers a particular provider (AesGcm prefers
+        // the non-FIPS BouncyCastle provider), that takes precedence.
+        if (preferredProvider != null) {
+            return Cipher.getInstance(algorithm, preferredProvider);
+        }
+
+        // Otherwise, if the user has specified a provider, go with that.
+        if (provider != null) {
+            return Cipher.getInstance(algorithm, preferredProvider);
```

Comment:
Is this correct? Seems like it should be `provider`, not `preferredProvider`

Reply:
Whoops! Nice catch!